src/supernova/circuit.rs

```diff
@@ -178,7 +178,15 @@ impl<'a, G: Group, SC: StepCircuit<G::Base>> SuperNovaAugmentedCircuit<'a, G, SC
     let program_counter = if self.params.is_primary_circuit {
       Some(AllocatedNum::alloc(
         cs.namespace(|| "program_counter"),
-        || Ok(self.inputs.get()?.program_counter),
+        || {
+          Ok(
+            self
+              .inputs
+              .get()?
+              .program_counter
+              .expect("program_counter missing"),
```

Comment:
Why a panic rather than `SynthesisError::AssignementMissing`?

Reply:
It's because of the extremely confusing convention of how `Option` is overloaded in the bellman line of circuit-generation (phrased thus because maybe bellpepper will eventually help clean this up). Specifically, if *inputs* are missing (meaning we are synthesizing a blank shape), then the `.get()?` on line 185 will indeed result in the `SynthesisError::AssignmentMissing`. However, if that doesn't happen and inputs have been supplied, then those inputs are either for a primary or secondary circuit. In the secondary case, the program counter will not be present. However, in this code branch, we are specifically handling a primary circuit — so we require the actual value for the witness.